react/error.tsx

```diff
@@ -48,23 +56,80 @@ class ErrorPage extends Component {
     }
   }
 
-  private extractErrorInfo(runtime: any): any {
+  private extractErrorInfo() {
     const {
       account = null,
       workspace = null,
       culture: { locale } = { locale: null },
       route: { path } = { path: null },
-      loadedDevices = [null],
+      loadedDevices = null,
       production = null,
-    } = runtime
+      runtimeAvailable,
+    } = this.getRuntimeInfo()
 
     return {
       admin_account: account,
       admin_workspace: workspace,
       admin_locale: locale,
       admin_path: path,
-      admin_device: loadedDevices[0],
+      admin_device: Array.isArray(loadedDevices)
+        ? loadedDevices[0]
+        : loadedDevices,
       admin_production: production,
+      admin_runtime_available: runtimeAvailable,
+    }
+  }
+
+  /**
+   * Gets the Runtime information from the global __RUNTIME__ object
+   * injected by Render Server, if available, otherwise infer the runtime
+   * information from the window.location.
+   */
+  private getRuntimeInfo = (): RuntimeInfo & { runtimeAvailable: boolean } => {
+    const runtime = window?.global?.__RUNTIME__ ?? this.inferRuntimeInfo()
+
+    const {
+      account = null,
+      workspace = null,
+      culture: { locale } = { locale: null },
+      route: { path } = { path: null },
+      loadedDevices = [null],
+      production = null,
+    } = runtime
+
+    return {
+      account,
+      workspace,
+      culture: { locale },
+      route: { path },
+      loadedDevices,
+      production,
+      runtimeAvailable: !!window?.global?.__RUNTIME__,
+    }
+  }
+
+  /**
+   * Infer the runtime information from the window.location object.
+   */
+  private inferRuntimeInfo(): Partial<RuntimeInfo> {
+    const { host = '' } = window.location
+    let account = ''
+    let workspace = ''
+
+    if (!host.includes('--')) {
+      account = host.split('.')[0]
+      workspace = 'master'
+    } else {
+      workspace = host.split('--')[0]
+      account = host.split('--')[1].split('.')[0]
+    }
+
+    const { pathname } = window.location
+
+    return {
+      account,
+      workspace,
+      route: { path: pathname },
     }
```

Comment:
@marcelovicentegc é válido usar o [navigator.language](https://developer.mozilla.org/en-US/docs/Web/API/Navigator/language) como um valor pro `culture`?

```suggestion
    return {
      account,
      workspace,
      culture: navigator.language,
      route: { path: pathname },
    }
```

Reply:
Excelente pergunta, @vmourac-vtex! Acho válido incluirmos um fallback uma vez que sabemos, também, quando essa informação vem do contexto do IO ou quando foi definida por nós. @lucasaarcoverde qual sua opinião?

Reply:
Feito pessoal!